### ISSUE TITLE
reflect-metadata causes a http://localhost:3000/crypto Failed to load resource: the server responded with a status of 404 (Not Found)

### ISSUE THREAD

**Issue by ovione** [rbuckton/ReflectDecorators#36]
hi Ron

I have a problem when I try to use reflect-metadata with angular2

I took the angular2 quickstart like https://angular.io/docs/ts/latest/quickstart.html#!#config-files.
I included code that uses Reflect.metadata and Reflect.getMetadata.
I have a Person and json that represents that person. 
I deserialize that  json into that Person with this code.
The json is displayed on screen and the deserialized Person is printed to the console.

I started from the quickstart angular2 example and I followed the two options:
to use its plunker
to clone it
Option 1: from the angular2 quickstart plunker
Starting from the angular2 quickstart plunker I forked it with your code and added a Person to deserialize and it works like a charm. !!!
You can see the forked plunker here:
http://plnkr.co/edit/GY0k2XqurtfXDjHkAPYV?p=preview

From this example its important to note that:
the transpilation happens in the browser.
there is no need to import 'reflect-metadata'; to use Reflect.metadata or the Reflect.metadata, for some reason Reflect is found I guess because 'reflect-metadata' is a transistive dependency used by angular2.
Option 2: from the clone

I added the same code as in the plunker and imported it into intellij.
I did an npm install and npm start.

After npm start The first error came up and the server could not start. The error is  TS2339: Property 'metadata' does not exist on type 'typeof Reflect' on the terminal console and in my intellij.
To try to solve this problem I added: 

npm install reflect-metadata --save to have it in my node_modules
in the main.ts I added import 'reflect-metadata';
After doing this, the error in the intellij disappeared and now it seems to find the Reflect.

when I do npm start the server starts but when I access the page it throws an error in the console with the message 
http://localhost:3000/reflect-metadata Failed to load resource: the server responded with a status of 404 (Not Found)

To bypass this error I modified the systemjs.config.js with the following lines:
  ```
var map = {
    'app':                        'app', // 'dist',

    '@angular':                   'node_modules/@angular',
    'angular2-in-memory-web-api': 'node_modules/angular2-in-memory-web-api',
    'rxjs':                       'node_modules/rxjs',
    'reflect-metadata':           'node_modules/reflect-metadata'
  };

  // packages tells the System loader how to load when no filename and/or no extension
  var packages = {
    'app':                        { main: 'main.js',  defaultExtension: 'js' },
    'rxjs':                       { defaultExtension: 'js' },
    'angular2-in-memory-web-api': { main: 'index.js', defaultExtension: 'js' },
    'reflect-metadata':{ main: 'Reflect.js', defaultExtension: 'js' }
  };
```
Then I did the npm start again and now I have the error in the chrome console
```
Error: patchProperty/desc.set/wrapFn@http://localhost:3000/node_modules/zone.js/dist/zone.js:769:27
	Zone</ZoneDelegate</ZoneDelegate.prototype.invokeTask@http://localhost:3000/node_modules/zone.js/dist/zone.js:356:24
	Zone</Zone</Zone.prototype.runTask@http://localhost:3000/node_modules/zone.js/dist/zone.js:256:29
	ZoneTask/this.invoke@http://localhost:3000/node_modules/zone.js/dist/zone.js:423:29
	
	Error loading http://localhost:3000/crypto as "crypto" from http://localhost:3000/node_modules/reflect-metadata/Reflect.js
```
 
And here I'm stuck :(
Maybe I should include the reflect-metadata in a different way or maybe I'm configuring the systemjs.config.js in the wrong way when I had 'reflect-metadata'.

thanks
Ovidio

**Comment by rbuckton** [comment]
I think SystemJS is statically analyzing Reflect.js and trying to fetch a module named "crypto" due to the `require` call in Reflect.js. The problem is that this is supposed to be under a feature test. 

A possible solution would involve fooling SystemJS into ignoring the `require` call by modifying the expression. Changing line 840 in the output javascript to the following may work:

```js
var nodeCrypto = isNode && (void 0, require)("crypto");
```

**Comment by rbuckton** [comment]
Would you be willing to test a possible fix? If so, try the following from the command line:

```
npm install git://github.com/rbuckton/ReflectDecorators.git#fixRequireCrypto
```

This should install the library from git rather than npm. If this addresses your issue, please reply in the comments and I will apply the fix to master.

**Comment by ovione** [comment]
hi Ron,

yes that worked !!!
now I don't have any more crypto requirements and its great :).

Do you know when it will be in master ?


thanks
Ovidio

**Comment by rbuckton** [comment]
This is fixed and published to npm in release 0.1.4.

**Comment by rbuckton** [comment]
@ovione, This fix has unfortunately broken a number of webpack users. I'm looking into a different fix for this. Would you be able to test it?

If so, please install the following version so that I can be sure reverting part of this change does not break your scenario.

```
npm install git://github.com/rbuckton/ReflectDecorators.git#fixWebpack
```

**Comment by ovione** [comment]
ok i will test it now and I let you know inmediately.

cheers
Ovidio

**Comment by ovione** [comment]
hi Ron

unfortunately for me its the same error as before.

The error actually doesn't block my application, its just displayed on the
screen on the first load.
I'm working with a team and they were all happy that the error was gone but
if you say that a fix is comming we could live this error as it seems non
blocking.

By the way the primeng last version displays a similar version to yours but
I don't know if they use reflect-metadata. But this one blocks the entire
application from loading.
So we are still using a prior version of primeng.

Let me know in any case what you decide and if the fix is staying or if you
will revert it so that the webpack users are unblocked.

thanks
Ovidio

**Comment by rbuckton** [comment]
Fixed in f77ed03
